com.ibm.wala.core/src/com/ibm/wala/types/Selector.java

```diff
@@ -24,8 +27,17 @@
 
   private final Descriptor descriptor;
   
+  private final String s;
+  
+  private static final Map<String,Selector> CACHE = new HashMap<String,Selector>();
```

Comment:
@msridhar I reverted the respective commit for now. It did not have noticable effects anyway.